File: v2/harvest_api_sample.rb
require "net/http"
require "json"

PERSONAL_ACCESS_TOKEN = ENV["HARVEST_ACCESS_TOKEN"]
ACCOUNT_ID = ENV["HARVEST_ACCOUNT_ID"]

uri = URI("https://api.harvestapp.com/v2/users/me")

Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
  request = Net::HTTP::Get.new uri
  request["User-Agent"] = "Ruby Harvest API Sample"
  request["Authorization"] = "Bearer #{PERSONAL_ACCESS_TOKEN}"
  request["Harvest-Account-ID"] = ACCOUNT_ID

  response = http.request request
  json_response = JSON.parse(response.body)

  puts JSON.pretty_generate(json_response)
end

File: v2/harvest_api_sample.js
const https = require("https");

const options = {
  protocol: "https:",
  hostname: "api.harvestapp.com",
  path: "/v2/users/me",
  headers: {
    "User-Agent": "Node.js Harvest API Sample",
    "Authorization": "Bearer " + process.env.HARVEST_ACCESS_TOKEN,
    "Harvest-Account-ID": process.env.HARVEST_ACCOUNT_ID
  }
}

https.get(options, (res) => {
  const { statusCode } = res;

  if (statusCode !== 200) {
    console.error(`Request failed with status: ${statusCode}`);
    return;
  }

  res.setEncoding('utf8');
  let rawData = '';
  res.on('data', (chunk) => { rawData += chunk; });
  res.on('end', () => {
    try {
      const parsedData = JSON.parse(rawData);
      console.log(parsedData);
    } catch (e) {
      console.error(e.message);
    }
  });
}).on('error', (e) => {
  console.error(`Got error: ${e.message}`);
});
